### ISSUE TITLE
'WebSocketConnectionClosedException: Connection is already closed.' when token is incorrect

### ISSUE THREAD

**Issue by jneeven** [attzonko/mmpy_bot#147]
**Describe the bug**
If the bot token is incorrect, the bot will produce logs as if it connected successfully:

```
[01/17/2021 18:27:43] Starting new HTTPS connection (1): chat.website.com:443
[01/17/2021 18:27:43] https://chat.website.com:443 "GET /api/v4/users/me HTTP/1.1" 200 623
[01/17/2021 18:27:43] connected to mattermost
[01/17/2021 18:27:43] loading plugin "mmpy_bot.plugins"
[01/17/2021 18:27:43] registered respond_to plugin "users_access" to "^admin$"
[01/17/2021 18:27:43] registered respond_to plugin "busy_reply" to "^busy|jobs$"
[01/17/2021 18:27:43] registered respond_to plugin "hello_reply" to "hello$"
[01/17/2021 18:27:43] registered listen_to plugin "hello_reply_formatting" to "hello_formatting$"
[01/17/2021 18:27:43] registered respond_to plugin "hello_reply_formatting" to "hello_formatting"
[01/17/2021 18:27:43] registered listen_to plugin "hello_send" to "hello$"
[01/17/2021 18:27:43] registered respond_to plugin "hello_decorators" to "hello_decorators"
[01/17/2021 18:27:43] registered listen_to plugin "hello_decorators" to "hello_decorators"
[01/17/2021 18:27:43] registered respond_to plugin "web_api_reply" to "hello_web_api"
[01/17/2021 18:27:43] registered listen_to plugin "hello_comment" to "hello_comment"
[01/17/2021 18:27:43] registered listen_to plugin "hello_react" to "hello_react"
[01/17/2021 18:27:43] registered listen_to plugin "hello_reply_threaded" to "hello_reply_threaded"
[01/17/2021 18:27:43] registered listen_to plugin "help_request" to "^!help$"
[01/17/2021 18:27:43] registered respond_to plugin "help_request" to "^!help$"
[01/17/2021 18:27:43] registered listen_to plugin "info_request" to "^!info$"
[01/17/2021 18:27:43] registered respond_to plugin "info_request" to "^!info$"
[01/17/2021 18:27:43] registered respond_to plugin "ping_reply" to "^ping$"
[01/17/2021 18:27:43] registered respond_to plugin "sleep_reply" to "sleep (.*)"
[01/17/2021 18:27:43] keep active thread started
[01/17/2021 18:27:43] job running thread started
```

However, it will never actually receive any data. This is because this produces a `WebSocketConnectionClosedException: Connection is already closed.'`, which is caught [in this line](https://github.com/attzonko/mmpy_bot/blob/master/mmpy_bot/mattermost.py#L279). Apparently `self.connect_websocket` still returns True.

**Expected behavior**
It would be nice if it checks if it either checks if the token is correct, or this error is properly caught and raised.

**Operating Environment (please complete the following information):**
 - OS: Ubuntu 20.04
 - Python Version: 3.8
 - Mattermost Version:  5.31.6
 - mmpy_bot Version:  1.3.4

**Comment by attzonko** [comment]
I agree this is a bug. You seem comfortable with the code, please feel free to submit a PR with your proposed solution.

**Comment by jneeven** [comment]
Closed by #153.
